packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -240,6 +240,48 @@ String _generatePluralMethod(Message message, String translationForMessage) {
     .replaceAll('@(none)\n', '');
 }
 
+String _generateSelectMethod(Message message, String translationForMessage) {
+  if (message.placeholders.isEmpty) {
+    throw L10nException(
```

Comment:
Can we add units tests for exception checks? These should go under `packages/flutter_tools/test/general.shard/generate_localizations_test.dart` and there should be some other tests you can go off of in there.

Reply:
Thanks for hinting, tests were added (with one fix).